Tests/DecodingTests.swift

```diff
@@ -110,6 +110,120 @@ class DecodingTests: XCTestCase {
         let result = request.transformData(objectJSON)
         XCTAssertNil(result.value) //Should fail because RootKeyArray json contains [MockObject], not a single MockObject
     }
+    
+    func test_AnyNetworkRequesWithRootDecodableKey_SuccessfullyDecodesChildElement() {
+        let request = AnyNetworkRequest<MockObject>(method: .get, url: NetworkRequestTestDefaults.defaultURL, rootDecodingKey: "root_key")
+        let objectJSON = loadedJSONData(fromFileNamed: "RootKeyObject")
+        let result = request.transformData(objectJSON)
+        XCTAssertNotNil(result.value)
+    }
+    
+    func test_AnyNetworkRequesWithRootDecodableKey_SuccessfullyDecodesChildElements() {
+        let request = AnyNetworkRequest<[MockObject]>(method: .get, url: NetworkRequestTestDefaults.defaultURL, rootDecodingKey: "root_key")
+        let objectJSON = loadedJSONData(fromFileNamed: "RootKeyArray")
+        let result = request.transformData(objectJSON)
+        XCTAssertNotNil(result.value)
+    }
+    
+    func test_AnyNetworkRequesWithRootDecodableKey_SuccessfullyDecodesChildElementWhenExtraJSONPresent() {
+        let request = AnyNetworkRequest<MockObject>(method: .get, url: NetworkRequestTestDefaults.defaultURL, rootDecodingKey: "root_key")
+        let objectJSON = loadedJSONData(fromFileNamed: "RootKeyObjectPlus")
+        let result = request.transformData(objectJSON)
+        XCTAssertNotNil(result.value)
+    }
+    
+    func test_AnyNetworkRequesWithRootDecodableKey_FailsWithIncorrectKey() {
+        let request = AnyNetworkRequest<MockObject>(method: .get, url: NetworkRequestTestDefaults.defaultURL, rootDecodingKey: "incorrectkey")
+        let objectJSON = loadedJSONData(fromFileNamed: "RootKeyObject")
+        let result = request.transformData(objectJSON)
+        XCTAssertNil(result.value)
+        XCTAssertTrue(result.error?.error is DecodingError)
```

Comment:
Just for a bit more of a complete test, is there any way we could check for the `valueNotFound` case? And similarly, for the test below, check for the `dataCorruptedError` case?